client/components/editor/teaching-elements/Assessment/TextSelection/Selector.vue

```diff
@@ -0,0 +1,87 @@
+<template>
+  <div
+    v-html="structure"
+    :class="{ editing: this.isEditing }"
+    @mouseup="e => isEditing && onMouseup(e)"
+    class="selector ql-editor">
+  </div>
+</template>
+
+<script>
+import {
+  generateStructure,
+  getSelectionRange,
+  getText,
+  isBlast,
+  toggleAttributes,
+  toggleClasses,
+  nodeMapper
+} from './helpers';
+import Selections from './Selections';
+
+export default {
+  props: {
+    correct: { type: Array, required: true },
+    isEditing: { type: Boolean, required: true },
+    content: { type: String, required: true }
+  },
+  data() {
+    const selections = new Selections(this.correct);
+    const processNode = selections.processNode.bind(selections);
+    return {
+      selections,
+      structure: generateStructure(this.content, (node, index) => {
+        return nodeMapper(processNode, node, index);
+      })
```

Comment:
Move this above to separate variable:
```js
const structure = generateStructure(...);
```
so you can do:
```js
return { selections, structure };
```
Also this is in fact an html string, `structure` implies more complex type?